frontend/src/metabase/dashboard/actions/data-fetching.js

```diff
@@ -432,6 +452,20 @@ export const fetchDashboardCardData = createThunkAction(
   },
 );
 
+export const fetchDashboardCardMetadata = createThunkAction(
+  FETCH_DASHBOARD_CARD_METADATA,
+  () => async (dispatch, getState) => {
+    const allDashCards = getDashboardComplete(getState()).ordered_cards;
+    const selectedTabId = getSelectedTabId(getState());
+
+    const cards = allDashCards.filter(
+      dc =>
+        selectedTabId !== undefined && dc.dashboard_tab_id === selectedTabId,
```

Comment:
What happens if a dashboard does not have tabs - `selectedTabId = undefined`? Will we not fetch metadata then, or is this case being handled somewhere else?